source/math/FloatRotation3D.ooc

```diff
@@ -0,0 +1,58 @@
+		rx := FloatTransform2D createXRotation(this x, zDistance)
+		ry := FloatTransform2D createYRotation(this y, zDistance)
+		rz := FloatTransform2D createZRotation(this z)
+		return rz * rx * ry // Yaw -> Pitch -> Roll
+	}
+	round: func -> This { This new(this x round(), this y round(), this z round()) }
+	ceiling: func -> This { This new(this x ceil(), this y ceil(), this z ceil()) }
+	floor: func -> This { This new(this x floor(), this y floor(), this z floor()) }
+	minimum: func (ceiling: This) -> This { This new(this x minimum(ceiling x), this y minimum(ceiling y), this z minimum(ceiling z)) }
+	maximum: func (floor: This) -> This { This new(this x maximum(floor x), this y maximum(floor y), this z maximum(floor z)) }
+	clamp: func ~point(floor, ceiling: This) -> This { This new(this x clamp(floor x, ceiling x), this y clamp(floor y, ceiling y), this z clamp(floor z, ceiling z)) }
+	clamp: func ~float(floor, ceiling: Float) -> This { This new(this x clamp(floor, ceiling), this y clamp(floor, ceiling), this z clamp(floor, ceiling)) }
+	operator + (other: This) -> This { This new(this x + other x, this y + other y, this z + other z) }
+	operator - (other: This) -> This { This new(this x - other x, this y - other y, this z - other z) }
+	operator - -> This { This new(-this x, -this y, -this z) }
+	operator * (other: This) -> This { This new(this x * other x, this y * other y, this z * other z) }
+	operator / (other: This) -> This { This new(this x / other x, this y / other y, this z / other z) }
+	operator * (other: Float) -> This { This new(this x * other, this y * other, this z * other) }
+	operator / (other: Float) -> This { This new(this x / other, this y / other, this z / other) }
+	operator == (other: This) -> Bool { this x == other x && this y == other y && this z == other z }
+	operator != (other: This) -> Bool { this x != other x || this y != other y || this z != other z }
+	operator < (other: This) -> Bool { this x < other x && this y < other y && this z < other z }
+	operator > (other: This) -> Bool { this x > other x && this y > other y && this z > other z}
+	operator <= (other: This) -> Bool { this x <= other x && this y <= other y && this z <= other z }
+	operator >= (other: This) -> Bool { this x >= other x && this y >= other y && this z >= other z }
+	toString: func -> String { "%.8f" formatFloat(this x) >> ", " & "%.8f" formatFloat(this y) >> ", " & "%.8f" formatFloat(this z) }
+}
+operator * (left: Float, right: FloatRotation3D) -> FloatRotation3D { FloatRotation3D new(left * right x, left * right y, left * right z) }
+operator / (left: Float, right: FloatRotation3D) -> FloatRotation3D { FloatRotation3D new(left / right x, left / right y, left / right z) }
+operator * (left: Int, right: FloatRotation3D) -> FloatRotation3D { FloatRotation3D new(left * right x, left * right y, left * right z) }
+operator / (left: Int, right: FloatRotation3D) -> FloatRotation3D { FloatRotation3D new(left / right x, left / right y, left / right z) }
```

Comment:
Thanks for the information. As you probably have realized, it hasn't been pushed yet.